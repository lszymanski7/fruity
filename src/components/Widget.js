import React from 'react'
import Item from './Item'

const Widget = (props) => {
    const title = 'Select fruits:'
    const { fruits, setFruits } = props

    return (
        <div className="container">
            <div>
			    <h3>{title}</h3>
            </div>
            {
                fruits.map((fruit) => { 
                    const key = fruit.name.toLowerCase().replace(' ', '_')
                    return (
                        <Item 
                            key={key} 
                            id={fruit.id} 
                            name={fruit.name} 
                            img={fruit.img} 
                            checked={fruit.checked} 
                            fruits={fruits} 
                            setFruits={setFruits}  
                        />
                    )
                })
            }
		</div>
    )
}

export { Widget as default }