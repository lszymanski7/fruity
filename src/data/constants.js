const ICONS = {
    apple: require('../assets/images/icons/apple.webp'),
    banana: require('../assets/images/icons/banana.webp'),
    blackberry: require('../assets/images/icons/blackberry.webp'),
    blueberries: require('../assets/images/icons/blueberries.webp'),
    cherries: require('../assets/images/icons/cherries.webp'),
    coconut: require('../assets/images/icons/coconut.webp'),
    dragon_fruit: require('../assets/images/icons/dragon-fruit.webp'),
    fig: require('../assets/images/icons/fig.webp'),
    grapes: require('../assets/images/icons/grapes.webp'),
    kiwi: require('../assets/images/icons/kiwi.webp'),
    lemon: require('../assets/images/icons/lemon.webp'),
    lychee: require('../assets/images/icons/lychee.webp'),
    mango: require('../assets/images/icons/mango.webp'),
    melon: require('../assets/images/icons/melon.webp'),
    orange: require('../assets/images/icons/orange.webp'),
    papaya: require('../assets/images/icons/papaya.webp'),
    peach: require('../assets/images/icons/peach.webp'),
    pear: require('../assets/images/icons/pear.webp'),
    pineapple: require('../assets/images/icons/pineapple.webp'),
    plum: require('../assets/images/icons/plum.webp'),
    pomegranate: require('../assets/images/icons/pomegranate.webp'),
    star_fruit: require('../assets/images/icons/star-fruit.webp'),
    strawberry: require('../assets/images/icons/strawberry.webp'),
    watermelon: require('../assets/images/icons/watermelon.webp')
}

const LOGOS = {
    fruity: {
        size_128x128: require('../assets/images/logos/fruity-128x128.webp'),
        size_256x256: require('../assets/images/logos/fruity-256x256.webp'),
        size_512x512: require('../assets/images/logos/fruity-512x512.webp')
    },
    github: require('../assets/images/logos/github.svg')
}

const SHAPES = {
    moon: require('../assets/images/shapes/moon.svg'),
    reset: require('../assets/images/shapes/reset.svg'),
    sun: require('../assets/images/shapes/sun.svg')
}

export { ICONS, LOGOS, SHAPES }
