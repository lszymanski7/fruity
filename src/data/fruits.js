import { ICONS } from './constants'

// Fruits data
const fruits = [
    {
        id: 1,
        name: 'apple',
        img: ICONS.apple,
        checked: false
    },
    {
        id: 2,
        name: 'banana',
        img: ICONS.banana,
        checked: false
    },
    {
        id: 3,
        name: 'blackberry',
        img: ICONS.blackberry,
        checked: false
    },
    {
        id: 4,
        name: 'blueberries',
        img: ICONS.blueberries,
        checked: false
    },
    {
        id: 5,
        name: 'cherries',
        img: ICONS.cherries,
        checked: false
    },
    {
        id: 6,
        name: 'coconut',
        img: ICONS.coconut,
        checked: false
    },
    {
        id: 7,
        name: 'dragon fruit',
        img: ICONS.dragon_fruit,
        checked: false
    },
    {
        id: 8,
        name: 'fig',
        img: ICONS.fig,
        checked: false
    },
    {
        id: 9,
        name: 'grapes',
        img: ICONS.grapes,
        checked: false
    },
    {
        id: 10,
        name: 'kiwi',
        img: ICONS.kiwi,
        checked: false
    },
    {
        id: 11,
        name: 'lemon',
        img: ICONS.lemon,
        checked: false
    },
    {
        id: 12,
        name: 'lychee',
        img: ICONS.lychee,
        checked: false
    },
    {
        id: 13,
        name: 'mango',
        img: ICONS.mango,
        checked: false
    },
    {
        id: 14,
        name: 'melon',
        img: ICONS.melon,
        checked: false
    },
    {
        id: 15,
        name: 'orange',
        img: ICONS.orange,
        checked: false
    },
    {
        id: 16,
        name: 'papaya',
        img: ICONS.papaya,
        checked: false
    },
    {
        id: 17,
        name: 'peach',
        img: ICONS.peach,
        checked: false
    },
    {
        id: 18,
        name: 'pear',
        img: ICONS.pear,
        checked: false
    },
    {
        id: 19,
        name: 'pineapple',
        img: ICONS.pineapple,
        checked: false
    },
    {
        id: 20,
        name: 'plum',
        img: ICONS.plum,
        checked: false
    },
    {
        id: 21,
        name: 'pomegranate',
        img: ICONS.pomegranate,
        checked: false
    },
    {
        id: 22,
        name: 'star fruit',
        img: ICONS.star_fruit,
        checked: false
    },
    {
        id: 23,
        name: 'strawberry',
        img: ICONS.strawberry,
        checked: false
    },
    {
        id: 24,
        name: 'watermelon',
        img: ICONS.watermelon,
        checked: false
    }
]

export { fruits as default }
