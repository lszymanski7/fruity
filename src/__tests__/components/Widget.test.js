import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Widget from '../../components/Widget'
import { fruits } from '../../data/constants'

const props = {
    data: [...fruits],
    handleCheck: jest.fn(),
    handleReset: jest.fn()
}

describe('Widget', () => {
    it('should be rendered correctly.', () => {
        const { container } = render(<Widget {...props} />)
        expect(container).toMatchSnapshot()
    })

	it('should handle clicking (Reset).', async () => {
        render(<Widget {...props} />)
        const button = screen.getByRole('button')
        await userEvent.click(button)
        expect(props.handleReset).toHaveBeenCalled()
    })

    it('should have the correct title.', () => {
        render(<Widget {...props} />)
        const title = 'Fruits to select:'
        const h2 = screen.getByRole('heading', { level: 2 })
        expect(h2).toHaveTextContent(title)
    })

    it('should have 24 elements.', () => {
    	render(<Widget {...props} />)
    	const items = screen.getAllByRole('checkbox')
    	expect(items).toHaveLength(24)
    })
})
