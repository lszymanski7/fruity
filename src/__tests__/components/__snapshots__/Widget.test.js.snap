// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget should be rendered correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="widget"
      >
        <div
          class="widget-header"
        >
          <h2>
            Fruits to select:
          </h2>
        </div>
        <div
          class="widget-container"
        >
          <div
            class="item"
          >
            <img
              alt="Apple"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="apple-color.png"
            />
            <p
              class="item__title"
            >
              Apple
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Banana"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="banana-color.png"
            />
            <p
              class="item__title"
            >
              Banana
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Blackberry"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="blackberry-color.png"
            />
            <p
              class="item__title"
            >
              Blackberry
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Blueberries"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="blueberries-color.png"
            />
            <p
              class="item__title"
            >
              Blueberries
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Cherries"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="cherries-color.png"
            />
            <p
              class="item__title"
            >
              Cherries
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Coconut"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="coconut-color.png"
            />
            <p
              class="item__title"
            >
              Coconut
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Dragon Fruit"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="dragon-fruit-color.png"
            />
            <p
              class="item__title"
            >
              Dragon Fruit
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Fig"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="fig-color.png"
            />
            <p
              class="item__title"
            >
              Fig
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Grapes"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="grapes-color.png"
            />
            <p
              class="item__title"
            >
              Grapes
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Kiwi"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="kiwi-color.png"
            />
            <p
              class="item__title"
            >
              Kiwi
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Lemon"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="lemon-color.png"
            />
            <p
              class="item__title"
            >
              Lemon
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Lychee"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="lychee-color.png"
            />
            <p
              class="item__title"
            >
              Lychee
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Mango"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="mango-color.png"
            />
            <p
              class="item__title"
            >
              Mango
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Melon"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="melon-color.png"
            />
            <p
              class="item__title"
            >
              Melon
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Orange"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="orange-color.png"
            />
            <p
              class="item__title"
            >
              Orange
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Papaya"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="papaya-color.png"
            />
            <p
              class="item__title"
            >
              Papaya
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Peach"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="peach-color.png"
            />
            <p
              class="item__title"
            >
              Peach
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Pear"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="pear-color.png"
            />
            <p
              class="item__title"
            >
              Pear
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Pineapple"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="pineapple-color.png"
            />
            <p
              class="item__title"
            >
              Pineapple
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Plum"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="plum-color.png"
            />
            <p
              class="item__title"
            >
              Plum
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Pomegranate"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="pomegranate-color.png"
            />
            <p
              class="item__title"
            >
              Pomegranate
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Star Fruit"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="star-fruit-color.png"
            />
            <p
              class="item__title"
            >
              Star Fruit
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Strawberry"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="strawberry-color.png"
            />
            <p
              class="item__title"
            >
              Strawberry
            </p>
          </div>
          <div
            class="item"
          >
            <img
              alt="Watermelon"
              class="item__image"
              id="bbb33b2d-e07d-4d13-a869-f3790e100477"
              src="watermelon-color.png"
            />
            <p
              class="item__title"
            >
              Watermelon
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="widget"
    >
      <div
        class="widget-header"
      >
        <h2>
          Fruits to select:
        </h2>
      </div>
      <div
        class="widget-container"
      >
        <div
          class="item"
        >
          <img
            alt="Apple"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="apple-color.png"
          />
          <p
            class="item__title"
          >
            Apple
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Banana"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="banana-color.png"
          />
          <p
            class="item__title"
          >
            Banana
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Blackberry"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="blackberry-color.png"
          />
          <p
            class="item__title"
          >
            Blackberry
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Blueberries"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="blueberries-color.png"
          />
          <p
            class="item__title"
          >
            Blueberries
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Cherries"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="cherries-color.png"
          />
          <p
            class="item__title"
          >
            Cherries
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Coconut"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="coconut-color.png"
          />
          <p
            class="item__title"
          >
            Coconut
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Dragon Fruit"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="dragon-fruit-color.png"
          />
          <p
            class="item__title"
          >
            Dragon Fruit
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Fig"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="fig-color.png"
          />
          <p
            class="item__title"
          >
            Fig
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Grapes"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="grapes-color.png"
          />
          <p
            class="item__title"
          >
            Grapes
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Kiwi"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="kiwi-color.png"
          />
          <p
            class="item__title"
          >
            Kiwi
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Lemon"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="lemon-color.png"
          />
          <p
            class="item__title"
          >
            Lemon
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Lychee"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="lychee-color.png"
          />
          <p
            class="item__title"
          >
            Lychee
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Mango"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="mango-color.png"
          />
          <p
            class="item__title"
          >
            Mango
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Melon"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="melon-color.png"
          />
          <p
            class="item__title"
          >
            Melon
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Orange"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="orange-color.png"
          />
          <p
            class="item__title"
          >
            Orange
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Papaya"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="papaya-color.png"
          />
          <p
            class="item__title"
          >
            Papaya
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Peach"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="peach-color.png"
          />
          <p
            class="item__title"
          >
            Peach
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Pear"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="pear-color.png"
          />
          <p
            class="item__title"
          >
            Pear
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Pineapple"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="pineapple-color.png"
          />
          <p
            class="item__title"
          >
            Pineapple
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Plum"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="plum-color.png"
          />
          <p
            class="item__title"
          >
            Plum
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Pomegranate"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="pomegranate-color.png"
          />
          <p
            class="item__title"
          >
            Pomegranate
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Star Fruit"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="star-fruit-color.png"
          />
          <p
            class="item__title"
          >
            Star Fruit
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Strawberry"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="strawberry-color.png"
          />
          <p
            class="item__title"
          >
            Strawberry
          </p>
        </div>
        <div
          class="item"
        >
          <img
            alt="Watermelon"
            class="item__image"
            id="bbb33b2d-e07d-4d13-a869-f3790e100477"
            src="watermelon-color.png"
          />
          <p
            class="item__title"
          >
            Watermelon
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
